### DISCUSSION TITLE
Compare MGIS and MTEST (python)

### DISCUSSION THREAD

**Discussion by NsmKsm** [thelfer/MFrontGenericInterfaceSupport#138] [General]
Hello, 

I am implementing an elasto-visco-plastic problem (for polycrystals) in python and I want to use Mfront as an integration function (for behaviour) through MGIS. In order the make sure that everything is correct, I picked the "Meric Cailletaud" model as an example and I compared the results generated by MGIS and MTEST. 

My example is found in this link:  https://github.com/NsmKsm/compare_mgis_mtest, 

The generated results from MGIS and MTEST are different and, sadly, I could not figure out what's happening. 

Thanks in advance, 
Best,

**Comment by thelfer** [comment]
Ok, an easy one :) You set the time increment to `0` in the `MGIS` case. With the following correction, everything is fine:

~~~~python
    dt = time[tn] - time[tn - 1]
    mgisbv.integrate(m, mit, dt, 0, m.n)
~~~~

**Reply by NsmKsm** [reply]
Thank you for fast response.
So the third argument is the time increment. What about the fourth ? 
Also, is there a way to impose mixed conditions (e.g. imposing strains and stresses to simulate tensile tests) ?

**Reply by thelfer** [reply]
> So the third argument is the time increment. What about the fourth ?

Fourth argument is the number of the first integration point. This way, jobs can be easily r parallelized.

> Also, is there a way to impose mixed conditions (e.g. imposing strains and stresses to simulate tensile tests) ?

In `MTest`, certainly.

In `MGIS`, this is not straightforward. `MGIS` is meant to be called by a FEA/FFT/MPM solver: it does not handle equilibrium out of the box and is not meant to.

**Reply by thelfer** [reply]
BTW, you may want to read the C++ documentation which details the meaning of each argument :)

**Reply by NsmKsm** [reply]
Thank you.
